packages/core/test/features/st-var.spec.ts

```diff
@@ -1259,162 +1284,569 @@ describe(`features/st-var`, () => {
+                expect(computedVars.map.value).to.eql({
+                    a: {
+                        b: 'red',
+                    },
+                });
+                expect(computedVars.map.input).to.eql({
+                    type: 'st-map',
+                    flatValue: undefined,
+                    value: {
+                        a: {
+                            type: 'st-map',
+                            flatValue: undefined,
+                            value: {
+                                b: {
+                                    flatValue: 'red',
+                                    type: 'string',
+                                    value: 'red',
+                                },
+                            },
                         },
-                        createValue: ([size, style, color]) => {
-                            return {
-                                size,
-                                style,
-                                color,
-                            };
+                    },
+                });
+            });
+
+            it('should get computed custom value st-var', () => {
+                const { stylable, sheets } = testStylableCore({
+                    '/entry.st.css': `
+                    @st-import [stBorder as createBorder] from './st-border.js';
+    
+                    :vars {
+                        border: createBorder(1px, solid, red);
+                    }
+                    `,
+                    // Stylable custom value
+                    '/st-border.js': stBorderDefinitionMock,
+                });
+
+                const { meta } = sheets['/entry.st.css'];
+                const computedVars = stylable.stVar.getComputed(meta);
+
+                expect(Object.keys(computedVars)).to.eql(['border']);
+                expect(computedVars.border).to.containSubset({
+                    value: '1px solid red',
+                    input: {
+                        type: 'createBorder',
+                        flatValue: '1px solid red',
+                        value: {
+                            color: 'red',
+                            size: '1px',
+                            style: 'solid',
                         },
-                        getValue: (value, index) => {
-                            return value[index];
+                    },
+                    diagnostics: { reports: [] },
+                });
+            });
+
+            it('should get deep computed custom value st-var', () => {
+                const { stylable, sheets } = testStylableCore({
+                    '/entry.st.css': `
+                    @st-import [stBorder] from './st-border.js';
+    
+                    :vars {
+                        array: st-array(blue, stBorder(1px, solid, blue));
+                        map: st-map(
+                                border stBorder(value(array, 1, size), 
```

Comment:
like breaks is off here. stBorder first param is inline and the other 2 are on their own lines.